mlir/tools/mlir-rocm-runner/mlir-rocm-runner.cpp

```diff
@@ -46,15 +47,26 @@ static cl::opt<std::string> features("feature", cl::desc("target features"),
                                      cl::value_desc("AMDGPU target features"),
                                      cl::init(""));
 
+static cl::opt<int> deviceNum(
+    "device", cl::desc("Device number"),
+    cl::value_desc(
+        "0-indexed number of the GPU to execute on for multiple-GPU systems"),
+    cl::init(0));
+
 static LogicalResult runMLIRPasses(ModuleOp m) {
   PassManager pm(m.getContext());
   applyPassManagerCLOptions(pm);
 
+  if (failed(switchToDevice(deviceNum.getValue()))) {
+    return failure();
+  }
```

Comment:
Refer to previous comment:
 - This can be its own component that deal with runtime
 - You can extract hip related calls outside from `BackendUtils`, and
 - In BackendUtils, it can assume whatever passed to it is valid

This way, `mlir-rocm-runner` will be dependent on hip runtime but `mlir-miopen-driver` will not.

Reply:
Good idea. Now part of me wants to be maximally safe and write this as a pass that slips a `hipSetDevice` into main(), which we then add in mlir-rocm-runner